File: Python/347_Top_K_Frequent_Elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        cnt={}
        for i in nums:
            if i not in cnt.keys():
                cnt[i]=0
            cnt[i]+=1

        sorted_counts=sorted(cnt.items(), key= lambda y:y[1],reverse=True)
        res=[]
        for i in range(k):
            res.append(sorted_counts[i][0])

        return res

File: Java/347_Top_K_Frequent_Elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer,Integer> freq= new HashMap();
        for(int i:nums){
            freq.put(i,freq.getOrDefault(i,0)+1);
        }
        
        PriorityQueue <Map.Entry<Integer,Integer>> minHeap=new PriorityQueue <>((a,b) -> a.getValue()-b.getValue());
        //add to minheap:
        for(Map.Entry<Integer,Integer> en: freq.entrySet()){
            minHeap.offer(en);
            if(minHeap.size()>k){
                minHeap.poll(); // remove the one with min freq
            }
        }

        int[] res= new int[k];
        int index=k-1;
        while(!minHeap.isEmpty()){
            res[index]=minHeap.poll().getKey();
            index--;
        }

        return res;
    }
}